course/templates/parsons_question.html

```diff
@@ -1,47 +1,103 @@
 {% extends 'base_question_view.html' %}
 
+{% load static %}
+
+{% block script %}
+{{ block.super}}
+<link rel="stylesheet" href="{% static 'course/css/parsons.css' %}"/>
+{% endblock %}
 
 {% block submit_form %}
 
-    <div class="container">
-        <div class="row">
-            <div class="col-md-6">
-                <h5>Lines</h5>
-                <div class="left-container">
-                    {% for line in uqj.get_lines %}
-                        <div class="container-object">{{ line }}</div>
-                    {% endfor %}
+<div class="container">
+    <div class="row">
+        <div class="col-md-6">
+            <h5>Lines</h5>
+            <div class="left-container">
+                {% for line in uqj.get_lines %}
+                <div class="container-object">
+                    <pre class="code_lines my-0">{{ line }}</pre>
                 </div>
+                {% endfor %}
             </div>
-            <div class="col-md-6">
-                <h5>Your Solution</h5>
-                <div class="right-container">
+        </div>
+        <div class="col-md-6">
+            <h5>Your Solution</h5>
+            <div class="right-container">
 
-                </div>
             </div>
         </div>
     </div>
-    <script>
-        drake = dragula([$(".left-container")[0], $('.right-container')[0]])
-            .on('drop', function (e1, target, source, sibling) {
-                calculate_source_code();
-            })
+</div>
+<script>
+    drake = dragula([$(".left-container")[0], $(".right-container")[0]])
+        .on("drop", function (e1, target, source, sibling) {
+            determine_indents();
+            remove_indents();
+            calculate_source_code();
+        });
 
-        function calculate_source_code() {
-            let code = $('.right-container').children().text();
-            $('#source_code').val(code);
-        }
-    </script>
-    <div class="my-3">
-        <form method="post" enctype="multipart/form-data">
-            {% csrf_token %}
+    function calculate_source_code() {
+        let parsons_lines = $(".right-container pre.code_lines");
+        let code = "";
+        parsons_lines.each(function (i, e) {
+            let line_value = $(e);
+            let line = line_value.text();
+            code += line + "\n";
+        });
+        $("#source_code").val(code);
+    }
 
-            <input type="hidden" id="source_code" name="code" class="form-control">
+    function determine_indents() {
+        let parsons_lines = $(".right-container pre.code_lines");
+        let count = 0;
+        parsons_lines.each(function (i, e) {
+            let line_value = $(e);
+            let line = line_value.text().trim();
```

Comment:
The lines are not trimmed in the backend, are they?
I think we have to trim it in the back-end too. and keep this just to be safe.

Reply:
now I see why you needed this.